packages/core/src/electron-main/electron-application.ts

```diff
@@ -0,0 +1,466 @@
+import * as path from 'path';
+import { Argv } from 'yargs';
+import { AddressInfo } from 'net';
+import { promises as fs } from 'fs';
+import { fork, ForkOptions } from 'child_process';
+import { FrontendApplicationConfig } from '@theia/application-package/lib/application-props';
+import URI from '../common/uri';
+import { FileUri } from '../node/file-uri';
+import { Deferred } from '../common/promise-util';
+import { MaybePromise } from '../common/types';
+import { ContributionProvider } from '../common/contribution-provider';
+import { ElectronSecurityToken } from '../electron-common/electron-token';
+const Storage = require('electron-store');
+const createYargs: (argv?: string[], cwd?: string) => Argv = require('yargs/yargs');
+
+/**
+ * Options passed to the main/default command handler.
+ */
+export interface MainCommandOptions {
+
+    /**
+     * By default, the first positional argument. Should be either a relative or absolute file-system path pointing to a file or a folder.
+     */
+    readonly file?: string;
+
+}
+
+/**
+ * Fields related to a launch event.
+ *
+ * This kind of event is triggered in two different contexts:
+ *  1. The app is launched for the first time, `secondInstance` is false.
+ *  2. The app is already running but user relaunches it, `secondInstance` is true.
+ */
+export interface ExecutionParams {
+    readonly secondInstance: boolean;
+    readonly argv: string[];
+    readonly cwd: string;
+}
+
+export const ElectronApplicationGlobals = Symbol('ElectronApplicationSettings');
+export interface ElectronApplicationGlobals {
+    readonly THEIA_APP_PROJECT_PATH: string
+    readonly THEIA_BACKEND_MAIN_PATH: string
+    readonly THEIA_FRONTEND_HTML_PATH: string
+}
+
+/**
+ * The default entrypoint will handle a very rudimentary CLI to open workspaces by doing `app path/to/workspace`. To override this behavior, you can extend and rebind the
+ * `ElectronApplication` class and overriding the `launch` method.
+ * A JSON-RPC communication between the Electron Main Process and the Renderer Processes is available: You can bind services using the `ElectronConnectionHandler` and
+ * `ElectronIpcConnectionProvider` APIs, example:
+ *
+ * From an `electron-main` module:
+ *
+ *     bind(ElectronConnectionHandler).toDynamicValue(context =>
+ *          new JsonRpcConnectionHandler(electronMainWindowServicePath,
+ *          () => context.container.get(ElectronMainWindowService))
+ *     ).inSingletonScope();
+ *
+ * And from the `electron-browser` module:
+ *
+ *     bind(ElectronMainWindowService).toDynamicValue(context =>
+ *          ElectronIpcConnectionProvider.createProxy(context.container, electronMainWindowServicePath)
+ *     ).inSingletonScope();
+ */
+export const ElectronMainContribution = Symbol('ElectronApplicationContribution');
+export interface ElectronMainContribution {
```

Comment:
please be consistent, i.e.
`FrontendApplication` then `FrontendApplicationContribution`, the same for backend.

But here:
- application is called `ElectronApplication`
- contribution is called `ElectronMainContribution`
- and symbol string is `ElectronApplicationContribution `

`ElectronApplication` -> `ElectronApplicationContribution` or 
`ElectronMainApplication` -> `ElectronMainApplicationContribution`

the point is that contribution provide lifecycle hooks for the application and should be named after it.

Reply:
> or
> `ElectronMainApplication` -> `ElectronMainApplicationContribution`

We use this.